src/main/java/org/jboss/ejb/client/ConfigurationBasedEJBClientContextSelector.java

```diff
@@ -96,23 +104,26 @@ private static void parseEJBClientType(final ConfigurationXMLStreamReader stream
         boolean gotInvocationTimeout = false;
         boolean gotGlobalInterceptors = false;
         boolean gotConnections = false;
+        boolean gotTransactionPropagation = false;
         for (;;) {
             final int next = streamReader.nextTag();
             if (next == START_ELEMENT) {
-                if (! streamReader.getNamespaceURI().equals(NS_EJB_CLIENT_3_0)) {
+                if (!recognizedNamespaces.contains(streamReader.getNamespaceURI())) {
                     throw streamReader.unexpectedElement();
                 }
                 final String localName = streamReader.getLocalName();
-                if (localName.equals("invocation-timeout") && ! gotInvocationTimeout) {
+                if (localName.equals("invocation-timeout") && !gotInvocationTimeout) {
                     gotInvocationTimeout = true;
                     parseInvocationTimeoutType(streamReader, builder);
-                }
-                else if (localName.equals("global-interceptors") && ! gotGlobalInterceptors) {
+                } else if (localName.equals("global-interceptors") && !gotGlobalInterceptors) {
                     gotGlobalInterceptors = true;
                     parseInterceptorsType(streamReader, builder);
-                } else if (localName.equals("connections") && ! gotConnections) {
+                } else if (localName.equals("connections") && !gotConnections) {
                     gotConnections = true;
                     parseConnectionsType(streamReader, builder);
+                } else if (localName.equals("transaction-propagation") && !gotTransactionPropagation) {
```

Comment:
We don't want transaction-propagation to be read if namespace is 3.0.
So, we need to read transaction-propagation only if the namespace is 3.1 Otherwise, an exception must be thrown.

Reply:
@fl4via thanks for the feedback! updated, is it better now?

Reply:
@kurobako sure thing it is!